SagaFlow.AspNetCore/Authentications/HttpContextUsernameProvider.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.AspNetCore.Http;
+using SagaFlow.Authentications;
+
+namespace SagaFlow.AspNetCore.Authentications;
+
+public class HttpContextUsernameProvider : IUsernameProvider
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
```

Comment:
I might be able to use this in my work for getting the Authorized user and Claims

Reply:
yeah, I reckon it would be something we would have to reconcile with your work this this one, I just made a super simple abstraction so it wouldn't interfer or easy to combine with what you have done.